### PR TITLE
feat(ByRole): Allow filter by value state

### PR DESCRIPTION
**Pull request by eps1lon**
Docs for https://github.com/testing-library/dom-testing-library/pull/1223

[Rendered preview](https://deploy-preview-1230--testing-library.netlify.app/docs/queries/byrole#value)

### CODE REVIEW

File: docs/queries/byrole.mdx [hunk 1]
```diff
@@ -321,6 +327,70 @@ To learn more about the `aria-level` property, see
...
+      aria-valuetext="medium"
+    >
+      Pitch
+    </button>
+  </section>
+</body>
+```
+
+you can query the `Volume` spinbutton using
+`getByRole('heading', { value: { now: 5 } })`.
```
Comment (timdeschryver) [inline docs/queries/byrole.mdx:367]: I think would be better to reword this to avoid confusion with the other snippet that has more queries.

```suggestion
you can query specific spinbutton(s) with the following queries,
```

File: docs/queries/byrole.mdx [hunk 2]
```diff
@@ -321,6 +327,70 @@ To learn more about the `aria-level` property, see
...
+
+```js
+getByRole('heading', {value: {now: 5}})
+// <button>Volume</button>
+
+getAllByRole('heading', {value: {min: 0}})
+// [
+//   <button>Volume</button>>,
```
Comment (timdeschryver) [inline docs/queries/byrole.mdx:375]:
```suggestion
//   <button>Volume</button>,
```

File: docs/queries/byrole.mdx [hunk 3]
```diff
@@ -321,6 +327,70 @@ To learn more about the `aria-level` property, see
 > The `level` option is _only_ applicable to the `heading` role. An error will
 > be thrown when used with any other role.
 
+### `value`
+
+A range widget can be queried by any value `getByRole('heading')` or by a
+specific value using the `level` option
+`getByRole('heading', { value: { now: 5, min: 0, max: 10, text: 'medium' } })`.
```
Comment (MatanBobi) [inline docs/queries/byrole.mdx:334]: Should this be `spinbutton`?
```suggestion
`getByRole('spinbutton', { value: { now: 5, min: 0, max: 10, text: 'medium' } })`.
```

**Review by timdeschryver — APPROVED** [review]
